include/revng/Pipeline/Step.h

```diff
@@ -62,6 +75,33 @@ class Step {
   const ContainerSet &containers() const { return Containers; }
   ContainerSet &containers() { return Containers; }
 
+  void setArtifacts(std::string ContainerName, const Kind *ArtifactsKind) {
+    Artifacts = ArtifactsInfo(std::move(ContainerName), ArtifactsKind);
+  }
+
+  const Kind *getArtifactsKind() {
+    if (Artifacts.isValid()) {
+      return Artifacts.Kind;
+    } else {
+      return nullptr;
+    }
+  }
+
+  ContainerSet::value_type *getArtifactsContainer() {
+    if (!Artifacts.isValid()) {
+      return nullptr;
+    }
+
+    auto &ContainerName = Artifacts.Container;
+    Containers[ContainerName];
+    auto ContainerIterator = Containers.find(ContainerName);
+    if (ContainerIterator != Containers.end()) {
```

Comment:
This doesn't make much sense. `operator[]` creates an entry, what's the point of looking it up afterwards?
We probably need to
```suggestion
    auto ContainerIterator = Containers.find(ContainerName);
    if (ContainerIterator != Containers.end()) {
```

Reply:
Fixed